perception/autoware_tensorrt_bevdet/src/bevdet.cpp

```diff
@@ -0,0 +1,717 @@
+  delete[] order;
+
+  std::vector<int> interval_starts_host;
+  std::vector<int> interval_lengths_host;
+
+  interval_starts_host.push_back(0);
+  int len = 1;
+  for (int i = 1; i < valid_feat_num_; i++) {
+    if (ranks_bev_host[i] != ranks_bev_host[i - 1]) {
+      interval_starts_host.push_back(i);
+      interval_lengths_host.push_back(len);
+      len = 1;
+    } else {
+      len++;
+    }
+  }
+
+  interval_lengths_host.push_back(len);
+  unique_bev_num_ = interval_lengths_host.size();
+
+  int * interval_starts_host_ptr = new int[interval_starts_host.size()];
+  int * interval_lengths_host_ptr = new int[interval_lengths_host.size()];
+
+  memcpy(
+    interval_starts_host_ptr, interval_starts_host.data(),
+    interval_starts_host.size() * sizeof(int));
+  memcpy(
+    interval_lengths_host_ptr, interval_lengths_host.data(),
+    interval_lengths_host.size() * sizeof(int));
+
+  ranks_bev_ptr.reset(ranks_bev_host);
+  ranks_depth_ptr.reset(ranks_depth_host);
+  ranks_feat_ptr.reset(ranks_feat_host);
+  interval_starts_ptr.reset(interval_starts_host_ptr);
+  interval_lengths_ptr.reset(interval_lengths_host_ptr);
+
+  RCLCPP_INFO(rclcpp::get_logger("BEVDet"), "valid_feat_num: %d", valid_feat_num_);
+  RCLCPP_INFO(rclcpp::get_logger("BEVDet"), "unique_bev_num: %d", unique_bev_num_);
+}
+
+void print_dim(nvinfer1::Dims dim, const std::string & name)
+{
+  std::ostringstream oss;
+  oss << name << " : ";
+  for (auto i = 0; i < dim.nbDims; i++) {
+    oss << dim.d[i] << ' ';
+  }
+  RCLCPP_INFO(rclcpp::get_logger("BEVDet"), "%s", oss.str().c_str());
+}
+
+int BEVDet::initEngine(const std::string & engine_file)
+{
+  if (deserializeTRTEngine(engine_file, &trt_engine_)) {
+    return EXIT_FAILURE;
+  }
+
+  if (trt_engine_ == nullptr) {
+    RCLCPP_ERROR(rclcpp::get_logger("BEVDet"), "Failed to deserialize engine file!");
+    return EXIT_FAILURE;
+  }
+  trt_context_ = trt_engine_->createExecutionContext();
+
+  if (trt_context_ == nullptr) {
+    RCLCPP_ERROR(rclcpp::get_logger("BEVDet"), "Failed to create TensorRT Execution Context!");
+    return EXIT_FAILURE;
+  }
+
+  // set bindings
+  std::vector<nvinfer1::Dims32> shapes{
+    {4, {n_img_, 3, src_img_h_, src_img_w_ / 4}},
+    {1, {3}},
+    {1, {3}},
+    {3, {1, n_img_, cam_params_size_}},
+    {1, {valid_feat_num_}},
+    {1, {valid_feat_num_}},
+    {1, {valid_feat_num_}},
+    {1, {unique_bev_num_}},
+    {1, {unique_bev_num_}},
+    {5, {1, adj_num_, bevpool_channel_, bev_h_, bev_w_}},
+    {3, {1, adj_num_, transform_size_}},
+    {2, {1, 1}}};
+
+  for (size_t i = 0; i < shapes.size(); i++) {
+    trt_context_->setBindingDimensions(i, shapes[i]);
+  }
+
+  buffer_map_.clear();
+  for (auto i = 0; i < trt_engine_->getNbBindings(); i++) {
+    auto dim = trt_context_->getBindingDimensions(i);
+    auto name = trt_engine_->getBindingName(i);
+    buffer_map_[name] = i;
+    print_dim(dim, name);
+  }
+
+  return EXIT_SUCCESS;
+}
+
+int BEVDet::deserializeTRTEngine(
```

Comment:
Namespaces: Wrap the function `deserializeTRTEngine` within the  `autoware_tensorrt_bevdet` namespace.